docs/enhancements/config-image-versions.md

```diff
@@ -0,0 +1,143 @@
+authors:
+  - "@lranjbar"
+creation-date: 2021-10-27
+last-updated: 2021-10-29
+---
+
+# Configurable Image Versions
+
+## Summary
+
+When deploying Assisted Installer a ConfigMap is required to do so. In this
+ConfigMap we have the values of various components of Assisted Installer including
+the images for the application. In the code for assisted-service, we define
+several environment variables for these images. We also provide default 
+values for these images which can differ from the desired value at runtime.
+
+For many container based software applications the version of the container ran 
+is determined at runtime. Consider the simple example of `docker run myrepo/myimage` 
+the docker client will resolve this to `hub.docker.com/myrepo/myimage:latest` and 
+then run the image on your system. The expectation is that image ran would be the 
+version you supplied using latest as the default tag.
+
+If we expand this example out to container orchestration technologies, like Openshift,
+the expectation is that the version supplied to Openshift would be the version 
+that is ran for all images and containers. The expectation is that if the version
+you supplied to Openshift doesn't exist then it would fail to pull the image and 
+throw the `ImagePullBackOff` error.
+
+## Motivation
+
+### Goals
+
+The main goal of this proposal is to have the ability to control the images used in 
+an Assisted Installer deployment using standard container deployment methods.
+
+### Non-Goals
+
+There are default versions of Openshift Container Platform and RHCOS images
+supplied to Assisted Installer. These are not the focus of this discussion only
+how we decide which images to deploy and interact with.
+
+## Proposal
+
+First, if we need to populate these environment variables in the code then 
+Assisted Installer should populate them with the environment variables given 
+in the ConfigMap at runtime.
+
+Second, have errors thrown when the environment variables for the images are 
+blank and not provided in the ConfigMap. If a user doesn't supply a value for a 
+required image we should give a runtime error for the user.
+
+Third, throw warnings (or errors) when the version in the environment variable 
+differs from what is supplied at runtime for Assisted Installer.
+
+Finally, it might be good to validate that all the version tags of Assisted 
+Installer match in a deployment. The versions of Assisted Installer are not 
+backwards compatible with each other.
+
+### User Stories
+
+#### Story 1
+
+As a user of Assisted Installer I expect the images provided in the ConfigMap
+to be the ones that are used by Openshift in the deployment.
+
+#### Story 2
+
+As a user of Assisted Installer I expect to have basic validations on missing
+values in the ConfigMap.
+
+#### Story 3
+
+As a user of Assisted Installer I expect to be warned if my image configuration
+drifts during runtime. (This could be a sign of instability of my installation.)
```

Comment:
This is basically trying to detect if the value in these environment variables have changed after Assisted Installer is already running. I'm still trying to think of how exactly we might get in this state, but essentially it seems to me like it would be not great. 🤔